python/GafferUI/CompoundNumericPlugValueWidget.py

```diff
@@ -120,7 +117,9 @@ def _convertValue( self, value ) :
 			value = value.value
 			if hasattr( value, "dimensions" ) and isinstance( value.dimensions(), int ) :
 				with self.getContext() :
-					result = self.getPlug().getValue()
+					result = sole( p.getValue() for p in self.getPlugs() )
+				if result is None :
```

Comment:
> Given that we're only using result for the dimensions of the converted value

We're not only using `result` for the dimensions of the converted value. I misread that too and reimplemented to use `defaultValue()` to avoid the compute. _That_ broke drag and drop. Actually what's going on here is that we're setting the value for any fields that exist in the dropped value, and leaving the remaining fields alone. So if you drag `0 1` onto `1 2 3` you get `0 1 3`.

> I think this would break dnd on plugs with different values.

It will, but given the semantics of `_convertValue()`, we can't return a single value that will keep everything intact for plugs that started with different values. I'm reluctant to spend more time on this at this point. This is a fairly esoteric interaction, and you can't drag+drop onto the HUD widgets yet anyway, because they're transient popups.

Reply:
Makes sense re: differing dimensions. I guess it seemed a shame that (unless I'm missing something) in prohibits operations for the other case, where the plugs _are_ all of the same dimension, but have different values.

Reply:
I'm pretty sure this code path only kicks in when the incoming value has a different dimension to the plugs themselves. So we still have the ability to drag V3f values onto multiple V3f plugs etc. Can revisit if it becomes problematic, but like I say, at first at least there will be no way to drag onto a floating multi-editor anyway...